specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/networkProfile.json

```diff
@@ -0,0 +1,486 @@
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "Network profile resource."
+    },
+    "NetworkProfileListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NetworkProfile"
+          },
+          "description": "A list of network profiles that exist in a resource group."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results."
+        }
+      },
+      "description": "Response for ListNetworkProfiles API service call."
+    },
+    "ContainerNetworkInterfacePropertiesFormat": {
+      "properties": {
+        "containerNetworkInterfaceConfiguration": {
+          "$ref": "#/definitions/ContainerNetworkInterfaceConfiguration",
+          "description": "Container network interface configuration from which this container network interface is created."
+        },
+        "container": {
+          "$ref": "#/definitions/Container",
+          "description": "Reference to the conatinaer to which this container network interface is attached."
+        }
+      }
+    },
+    "ContainerNetworkInterface": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ContainerNetworkInterfacePropertiesFormat",
+          "description": "Container network interface properties."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Container network interface child resource."
+    },
+    "ContainerNetworkInterfaceConfigurationPropertiesFormat": {
+      "properties": {
+        "ipConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IPConfigurationProfile"
+          },
+          "description": "A list of ip configurations of the container network interface configuration."
+        },
+        "containerNetworkInterfaces": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerNetworkInterface"
+          },
+          "description": "A list of container network interfaces created from this container network interface configuration."
+        }
+      },
+      "description": "Container network interface configuration properties."
+    },
+    "ContainerNetworkInterfaceConfiguration": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ContainerNetworkInterfaceConfigurationPropertiesFormat",
```

Comment:
You probably want to use x-ms-client-flatten to make better SDK.
Example:
https://github.com/Azure/azure-rest-api-specs/blob/32b0d873aa851d456dfde7d6ba1d89ff33f897f0/specification/network/resource-manager/Microsoft.Network/stable/2016-12-01/applicationGateway.json#L511